src/woocommerce/helpers.ts

```diff
@@ -0,0 +1,182 @@
+import http, { Response } from 'k6/http'
+import { parseHTML } from 'k6/html'
+
+import papaparse, { ParseResult } from 'papaparse'
+import slug from 'slugify'
+
+import {
+    map, flatMap, filter, get, uniq, parseInt,
+    replace, split, trim, toString, toLower,
+    has, isEmpty, isString
+} from 'lodash'
+
+
+const SHOP_URL = __ENV.SHOP_URL || 'http://localhost:8080/'
+
+//
+//  SAMPLE DATA
+
+export type Product = {
+    id: number
+    sku: string
+    name: string
+    categories: string[]
+}
+
+function isValidProduct(productData: object) {
+    return (
+        !isEmpty(get(productData, '\ufeffID'))
+        && get(productData, 'Type') === 'simple'
+        && get(productData, 'Published') === '1'
+        && get(productData, 'In stock?') === '1'
+    )
+}
+
+function toProduct(productData: object): Product {
+    return {
+        id: parseInt(get(productData, '\ufeffID')),
+        sku: get(productData, 'SKU'),
+        name: get(productData, 'Name'),
+        categories: map(split(get(productData, 'Categories'), ' > '), toLower)
+    }
+}
+
+export function loadSeedData() {
+    const parsedData: ParseResult<object> = papaparse.parse(open('../sample_products.csv'), { header: true })
+    const filteredData = filter(parsedData.data, isValidProduct)
+    const products = map(filteredData, toProduct)
+    const categories = uniq(flatMap(products, 'categories'))
+
+    return {
+        products,
+        categories
+    }
+}
+
+
+//
+//  RESPONSE CHECKS
+
+export function isOK(response: Response): boolean {
+    return response.status === 200
+}
+
+export function hasCookie(name: string): (response: Response) => boolean {
+    return (response: Response) => {
+        return has(response.cookies, name)
+    }
+}
+
+export function hasElementWithText(selector: string, text: string): (response: Response) => boolean {
+    return (response: Response) => {
+        if (!isString(response.body)) {
+            return false
+        }
+
+        const doc = parseHTML(response.body)
+        const element = doc.find(selector)
+        return element && trim(element.text()) === text
+    }
+}
+
+
+//
+//  URL BUILDERS
+
+export function shopURL(path: string) {
+    const pathname = path[0] === '/' ? path : `/${path}`
+    const baseURL = replace(SHOP_URL, /\/$/, '')
+    return `${baseURL}${pathname}`
+}
+
+export function productURL(product: Product) {
+    return shopURL(`/product/${slug(product.name)}`)
+}
+
+export function categoryURL(category: string) {
+    return shopURL(`/product-category/${category}`)
+}
+
+export function cartURL() {
+    return shopURL('/cart')
+}
+
+export function checkoutURL() {
+    return shopURL('/checkout')
+}
+
+export function ajaxURL(method: string) {
+    return shopURL(`/?wc-ajax=${method}`)
+}
+
+
+
+
+//
+//  NAVIGATION
+
+export function openProductPage(product: Product): Response {
+    const response = http.get(productURL(product))
+    check(response, { isOK })
+    sleep(1)
```

Comment:
Is it this function's concern to disrupt/control the flow?
Maybe one will want fine control over the flow (openProductPage -> openCart -> sleep) or wants a smaller sleep timeout. Maybe we can use callbacks or have this sleep configurable.

Reply:
The sleep is added as per `k6`s recommandation in order to simulate a more realistic browsing pattern.  
I'm not really fond of it.  
Maybe we could remove it altogether.  
Or just remove it from all helpers and add it in the test file, between steps, if we decide to keep it.